### DISCUSSION TITLE
Specifying spatially and temporally varying flow and solute transport boundary conditions

### DISCUSSION THREAD

**Discussion by jwheiss** [idaholab/moose#19936] [Q&A General] [answered]
Hello,

Is it possible to specify spatially and temporally varying flow and solute transport boundary conditions in PorousFlow? For instance, if pressure and solute concentration vary spatially and temporally along the boundary, how do we specify that boundary condition?

The time-varying behavior of the pressure and solute at the boundary are complex and cannot be represented with a function. Can the user supply an external file that specifies over time the pressure and solute concentration at each element or at a specified interval along a boundary? If so, how would that be implemented? The spatially and temporally varying specified pressure also needs to be coupled to solid mechanics (coupling_type = HydroMechanical).

Thanks,
James

**Comment by WilkAndy** [comment, marked as answer]
Hi James,

See the PorousFlow [boundary](https://mooseframework.inl.gov/modules/porous_flow/boundaries.html) page.  I suggest using a [PorousFlowPiecewiseLinearSink](https://mooseframework.inl.gov/source/bcs/PorousFlowPiecewiseLinearSink.html) or [FunctionDirichletBC](https://mooseframework.inl.gov/source/bcs/FunctionDirichletBC.html).  In these cases the `flux_function` or `function` can be specified by your file using, for instance, a [PiecewiseMulticonstant](https://mooseframework.inl.gov/source/functions/PiecewiseMulticonstant.html) function.

**Reply by jwheiss** [reply]
Hi Andy,

Thanks very much for your help. I'll take a look at the functions you mentioned.

Best,
James
